pkg/cache/cache.go

```diff
@@ -118,7 +118,19 @@ func New(defaultAudience, prefix string, clientset kubernetes.Interface) Service
 				c.addSA(sa)
 			},
 			DeleteFunc: func(obj interface{}) {
-				sa := obj.(*v1.ServiceAccount)
+				sa, ok := obj.(*v1.ServiceAccount)
+				if !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
+						return
+					}
+					pod, ok = tombstone.Obj.(*v1.ServiceAccount)
```

Comment:
should be `sa` not `pod` right?

Reply:
yep, fixed, goimports quirk also (missing import)